conda/gateways/connection/session.py

```diff
@@ -78,19 +113,21 @@ def __init__(self):
 
         else:
             # Configure retries
-            retry = Retry(total=context.remote_max_retries,
-                          backoff_factor=context.remote_backoff_factor,
-                          status_forcelist=[413, 429, 500, 503],
-                          raise_on_status=False)
-            http_adapter = HTTPAdapter(max_retries=retry)
+            retry = Retry(
+                total=context.remote_max_retries,
+                backoff_factor=context.remote_backoff_factor,
+                status_forcelist=[413, 429, 500, 503],
+                raise_on_status=False,
+            )
+            http_adapter = HTTPAdapter(max_retries=retry, ssl_context=self.ssl_context)
```

Comment:
3. pip's HTTPAdapter (import changed at top of file) uses a mixin class to add the ssl_context constructor argument, and pass it down the stack to where it needs to go. We should have our own implementation of "specify the ssl_context" instead of using pip's.